src/jit/codegenarmarch.cpp

```diff
@@ -962,6 +962,14 @@ void CodeGen::genPutArgSplit(GenTreePutArgSplit* treeNode)
         // Generate code to load the address that we need into a register
         genConsumeAddress(addrNode);
         addrReg = addrNode->gtRegNum;
+
+        // If addrReg equal to baseReg, we use the last target register as alternative baseReg.
+        // Because baseReg is not allocated as one of target register by masking,
+        // we can avoid baseReg, addrReg, and the last target register are all same.
```

Comment:
This comment is a little confusing. I would suggest:
```
        // If addrReg is equal to baseReg, we use the last target register as an alternative baseReg.
        // Because the candidate mask for the internal baseReg does not include any of the target registers,
        // we can ensure that baseReg, addrReg, and the last target register are not all the same.
```